src/components/AuthoritiesLookup/AuthoritiesLookup.js

```diff
@@ -59,38 +58,33 @@ const AuthoritiesLookup = ({
   const [isFilterPaneVisible, setIsFilterPaneVisible] = useState(true);
   const [showDetailView, setShowDetailView] = useState(false);
 
-  const { navigationSegmentValue, filters } = useContext(AuthoritiesSearchContext);
+  const { filters } = useContext(AuthoritiesSearchContext);
   const [, setSelectedAuthorityRecordContext] = useContext(SelectedAuthorityRecordContext);
 
   const toggleFilterPane = () => setIsFilterPaneVisible(!isFilterPaneVisible);
 
   const closeDetailView = () => {
     setShowDetailView(false);
-    setSelectedAuthorityRecordContext(null);
+  };
+
+  const openDetailView = () => {
+    setSelectedAuthorityRecordContext(authorities[0]);
+    setShowDetailView(true);
   };
 
   useEffect(() => {
     closeDetailView();
     // eslint-disable-next-line
   }, [filters]);
 
-  useEffect(() => {
-    const isDetailViewNeedsToBeOpened = getIsDetailViewNeedsToBeOpened(totalRecords, authorities[0], navigationSegmentValue);
-
-    if (isDetailViewNeedsToBeOpened) {
-      setSelectedAuthorityRecordContext(authorities[0]);
-      setShowDetailView(true);
-    }
-    // eslint-disable-next-line
-  }, [totalRecords, authorities, navigationSegmentValue]);
-
   const handleSubmitSearch = (e, ...rest) => {
     if (e?.preventDefault) {
       e.preventDefault();
       e.stopPropagation();
     }
 
     closeDetailView();
+    setSelectedAuthorityRecordContext(null);
```

Comment:
Can we move this call to `onSubmitSearch`? There are other calls to change context value and maybe we should keep them together

Reply:
Thanks, fixed